### ISSUE TITLE
Inconsistent precision format between Kornia and Torchvision

### ISSUE THREAD

**Issue by shijianjian** [kornia/kornia#490]
Since the Torchvision is heavily using PIL which is uint8 formatted, the output can hardly be reproduced using our current precision. Also, we may not want to downgrade our precision levels. In this case, I suggest having a global variable like ```PIL_MODE``` or so to be able to control Kornia precisions.

I found we have to update many basic functions first before testing on Kornia augmentation and Torchvision.

### to_grayscale
In Kornia, we have
```python
0.299 * r + 0.587 * g + 0.114 * b
```
but we need to update it to:
```python
torch.round(0.299 * 255 * r + 0.587 * 255 * g + 0.114 * 255 * b).int
```
While the torch.round is a big "buggy" to us. For example, torch.round(4.500) output 4 but it should be 5 in PIL, which can highly impede us from reproducing the same results.

### rgb_to_hsv
Haven't found easy fix.
### hsv_to_rgb
Haven't found easy fix.

I think issues like #489 is highly related to this precision problems.

**Comment by edgarriba** [comment]
@shijianjian this was one of the principle designs; to support only floating precision. Couple of reasons: 1) library target was meant for training 2) referring to 1, most of the low level operators are only supported and optimized for floating point precision.

If it's jut the case for data augmentation and to match a third-party lib (in this case PIL), I would make a workaround in such functionalities, or just simply add specifications in the documentation.

**Comment by ducha-aiki** [comment]
As we rely on pytorch on all operations and it has only float and half floats, I see no reason to have uint8 operations here. Moreover, they are not differentiable.

**Comment by shijianjian** [comment]
I understand that we are in a great advantage with the floating-point support. Also, I agree that uint8 is not differentiable. But if we wish to reproduce the exact same results with Torchvision, we definitely need to use many torch.round() to replicate what happened inside PIL.

**Comment by edgarriba** [comment]
I guess then it's a matter of testing, not for the library core. We can just have utility functions or is there any other thing that could be affected by this issue?

**Comment by shijianjian** [comment]
Take the example of the to_grayscale function.

from (Kornia)
```python
0.299 * r + 0.587 * g + 0.114 * b
```
to (PIL)
```python
torch.round(0.299 * 255 * r + 0.587 * 255 * g + 0.114 * 255 * b).int() / 255.
```

If we are not touching our original implementation, we definitely cannot reproduce the same results. One thing I am thinking is to have a switching parameter like ```PIL_MODE``` to control the function output. Do you have any suggestions for this?

**Comment by edgarriba** [comment]
have you measured the actual difference, because if not that large, we could assume correctness in our implementation

**Comment by shijianjian** [comment]
From what I have tested:
- adjust_hue
- adjust_contrast
- adjust_brightness
- adjust_saturation
- to_grayscale
- rgb_to_hsv
- hsv_to_rgb

Apart from the adjust_contrast and adjust_brightness, I am sure that we are doing correctly. But in terms of testing, I do not think it is doable in the current stage.

Also, we may be handling rgb_to_hsv and hsv_to_rgb differently comparing to PIL, but I think we are at least self-contained here.
